src/components/Modals/TradeTokens.js

```diff
@@ -0,0 +1,269 @@
+import cx from 'classnames';
+import _ from 'lodash';
+import { BN } from '@polkadot/util';
+import ModalRoot from './ModalRoot';
+import { TextInput } from '../InputComponents';
+import styles from './styles.module.scss';
+import Button from '../Button/Button';
+import { dexActions, walletActions } from '../../redux/actions';
+import { userSelectors, dexSelectors, walletSelectors } from '../../redux/selectors';
+import {
+  calculateAmountDesiredFormatted,
+  converTransferData,
+  convertToEnumDex,
+  formatProperlyValue,
+  formatterDecimals,
+  getBNDataToFindPrice,
+  getDecimalsBN,
+  getDecimalsForAsset,
+} from '../../utils/dexFormater';
+import { AssetsPropTypes } from '../Wallet/Exchange/proptypes';
+import { getSwapPriceExactTokensForTokens, getSwapPriceTokensForExactTokens } from '../../api/nodeRpcCall';
+
+function TradeTokensModal({
+  handleModal, assets, isBuy,
+}) {
+  const dispatch = useDispatch();
+  const walletAddress = useSelector(userSelectors.selectWalletAddress);
+  const assetBalance = useSelector(walletSelectors.selectorAssetBalance);
+  const reserves = useSelector(dexSelectors.selectorReserves);
+  const [isChecked, setIsChecked] = useState(false);
+  const isDisplayNone = isChecked ? null : styles.displayNone;
+  const [searchAmount, setSearchAmount] = useState('');
+  const [actualPrice, setActualPrice] = useState(null);
+
+  const {
+    asset1,
+    asset2,
+    assetData1,
+    assetData2,
+    asset1ToShow,
+    asset2ToShow,
+  } = assets;
+  const decimals1 = getDecimalsForAsset(asset1, assetData1?.decimals);
+  const decimals2 = getDecimalsForAsset(asset2, assetData2?.decimals);
+
+  const reservesThisAssets = useMemo(() => {
+    if (reserves && asset1 && asset2 && reserves[asset1 + asset2]) {
+      const indexAsset = asset1 + asset2;
+      const asset1Value = Number(reserves[indexAsset].asset1);
+      const asset2Value = Number(reserves[indexAsset].asset2);
+      return { ...reserves[indexAsset], asset1: asset1Value, asset2: asset2Value };
+    }
+    return null;
+  }, [asset1, asset2, reserves]);
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isValid },
+  } = useForm({
+    mode: 'onChange',
+    defaultValues: {
+      cos: null,
+    },
+  });
+
+  const onSubmit = async (data) => {
+    if (!isValid) return;
+    const { amountIn, minAmountPercent } = data;
+    const amountOut = minAmountPercent.length > 0 ? minAmountPercent : undefined;
+    try {
+      const { amount, amountMin } = await converTransferData(
+        asset1,
+        decimals1,
+        asset2,
+        decimals2,
+        amountIn,
+        isBuy,
+        amountOut,
+      );
+      const swapData = {
+        path: { asset1, asset2 },
+        amount,
+        amountMin,
+        sendTo: walletAddress,
+      };
+      dispatch(
+        isBuy
+          ? dexActions.swapTokensForExactTokens.call(swapData) : dexActions.swapExactTokensForTokens.call(swapData),
```

Comment:
I think there might be some misunderstanding here.
Both this calls do the same kind of trade - sell asset1 in exchange for asset2. It's just that with `swapTokensForExactTokens` is then you're saying "I sell asset1, I get 100 asset2" and `swapExactTokensForTokens` is "I sell 100 asset1, I get asset2". To actually sell asset2 and buy asset1 you need to swap them in the `path`.

Right now UI shows only "Buy LLD for Asset" and "Sell LLD For Asset". This isn't enough. In general I think that using custom buttons for this plainly sux UX-wise and even if us developers get confused our users have no chance and we should just copy Uniswap, buuut if we want to keep buttons, then we need 4 of them (there's a Polish CEX (zonda) that kinda does it like that, though with tabs instead of buttons):
* `Buy LLD for Asset` - swapTokensForExactTokens, where Asset is asset1 and LLD is asset2
* `Buy Asset for LLD` - swapTokensForExactTokens, where LLD is asset1 and Asset is asset2
* `Sell LLD for Asset` - swapExactTokensForTokens, where LLD is asset1 and Asset is asset2
* `Sell Asset for LLD` - swapExactTokensForTokens, where Asset is asset1 and LLD is asset2

In other words, `asset1` is always what user will lose, `asset2` is what user will gain.  Whether we use swapTokensForExactTokens or swapExactTokensForTokens depends on which amount user wants to specify, but doesn't affect the direction of the trade.

@DorianSternVukotic this is why you can't fix the ratio.

Reply:
After signal chat, we keep 2 buttons "Buy LLD for Asset" and "Sell LLD for Asset" and they decide the direction of trade (which asset is asset1 and which is asset2). For the forms, we do two inputs: one for how much we pay (asset1) and one for how much we receive (asset2). Then, depending on which input the user puts amount in, we decide on which side is exact (pay or receive). So:
1. `Buy LLD for Asset` - Asset is asset1, LLD is asset2
2. `Sell LLD for Asset` - LLD is asset1, Asset is asset2
3. If user puts amount in the first input (the "I lose/pay with" input), we do `swapExactTokensForTokens`
4. If user puts amount in the second input (the "I buy/receive"), we do `swapTokensForExactTokens`